kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -99,6 +137,21 @@ private[spark] class KubernetesClusterSchedulerBackend(
     // create a single k8s executor pod.
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
+
+    val submitArgs = mutable.ArrayBuffer.empty[String]
+
+    if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
+      submitArgs ++= Vector(
+        "dynamic-executors")
```

Comment:
How about just set a environment variable? An argument makes it hard to realize what this is really for, at least we can have a descriptive name like "SPARK_LAUNCH_SHUFFLE_SERVICE=1"